File: cpp/bc1188.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int main()
{
	double cVal[12][12], sum = 0.0;
	int fPos = 5, lPos = 6;
	char op;
	
	cin >> op;
	
	for(int i = 0; i < 12; i++)
	{
		for(int j = 0; j < 12; j++)
			cin >> cVal[i][j];
	}
	
	for(int i = 7; i < 12; i++)
	{
		for(int j = fPos; j <= lPos; j++)
			sum += cVal[i][j];
		
		fPos--;
		lPos++;
	}
	
	if(op != 'S') sum /= 30.0;
	
	cout << fixed << setprecision(1) << sum << endl;
}

File: python/bc1188.py
ma = [[None for j in range(12)] for i in range(12)]
res = 0.0
fPos = 5
lPos = 6

op = input()

for i in range(0, 12):
    for j in range(0, 12):
        ma[i][j] = float(input())

for i in range(7, 12):
    for j in range(fPos, lPos+1):
        res += ma[i][j]
    fPos -= 1
    lPos += 1

if op != "S": res /= 30.0

print(f"{res:.1f}")